olp-cpp-sdk-dataservice-read/src/ApiClientLookup.cpp

```diff
@@ -58,40 +58,46 @@ ApiClientLookup::ApiClientResponse ApiClientLookup::LookupApi(
     client::CancellationContext cancellation_context, std::string service,
     std::string service_version, FetchOptions options,
     client::OlpClientSettings settings) {
+  using namespace client;
   repository::ApiCacheRepository repository(catalog, settings.cache);
+  const auto hrn = catalog.ToCatalogHRNString();
 
   if (options != OnlineOnly && options != CacheWithUpdate) {
     auto url = repository.Get(service, service_version);
     if (url) {
-      OLP_SDK_LOG_INFO_F(kLogTag, "LookupApi(%s, %s) -> from cache",
-                         service.c_str(), service_version.c_str());
-      client::OlpClient client;
+      OLP_SDK_LOG_DEBUG_F(kLogTag, "LookupApi(%s/%s) found in cache, hrn='%s'",
+                          service.c_str(), service_version.c_str(),
+                          hrn.c_str());
+      OlpClient client;
       client.SetSettings(std::move(settings));
       client.SetBaseUrl(*url);
       return client;
     } else if (options == CacheOnly) {
-      return client::ApiError(
-          client::ErrorCode::NotFound,
-          "Cache only resource not found in cache (loopup api).");
+      return {{ErrorCode::NotFound, "CacheOnly: resource not found in cache"}};
     }
   }
 
+  OLP_SDK_LOG_INFO_F(kLogTag,
```

Comment:
Trace/Debug?

Reply:
I think info is good so that we see how often we request API lookup. Ideally, it is only once and then stored to cache.